### DISCUSSION TITLE
How to define another IP and port as 127.0.0.1:8080?

### DISCUSSION THREAD

**Discussion by carliedu** [clue/framework-x#227] [Q&A]
As I so far has seen, I could not find out how to run the web server an an other IP and port as 127.0.0.1 and 8080. How to pass another values as parameters on new FrameworkX\App(); ?

**Comment by carliedu** [comment]
I accidentally found an example from page https://framework-x.org/#simple that I would never have thought of: 
$ php -S localhost:8081 examples/index.php.

**Reply by SimonFrings** [reply]
@carliedu You can use  `$ php -S localhost:8081 examples/index.php` to run [PHP's built-in development web server](https://framework-x.org/docs/best-practices/deployment/#php-development-web-server) on your address of choice.

If you want to get the most out of Framework X you can use [Framework X's built-in web server](https://framework-x.org/docs/best-practices/deployment/#built-in-web-server), which provides a non-blocking implementation that can handle thousands of incoming connections and provide a much better user experience in high-load scenarios.

If you want to change the listen address of Framework X's built-in web server you can either [change the listen address](https://framework-x.org/docs/best-practices/deployment/#listen-address) through the X_LISTEN environment variable like this: `X_LISTEN=127.0.0.1:8081 php public/index.php` or by [configuring the dependency injection container](https://framework-x.org/docs/best-practices/controllers/#container-configuration) inside your `index.php`:
```php
$container = new Container([
    'X_LISTEN' => fn(string $PORT = '8080') => '0.0.0.0:' . $PORT
]);

$app = new App($container);
```

Reference: #38, #92 and others
